### PR TITLE
v1alpha3

### PR DESCRIPTION
**Pull request by njuettner**
Because I rolled out `v1alpha3` GS AWS CRD's we also need to ensure `aws-admission-controller` will look on this API version. CRD conversion happens before mutating and validating so we do not need `v1alpha2` anymore, because stored version is set to `v1alpha3` in all AWS installations.

Unfortunately this PR is a bit bigger because of v1alpha3 changes. I also added **infraRef** mutation for an upcoming AWS release. This is needed because we would still have `cluster`, `g8scontrolplane` and `machinedeployment` pointing to `infraRef` with version `v1alpha2`. I added also tests for that.

We only need to align and specify which version we want to update `infraRef` to `v1alpha3`.

This can be rolled out before we update https://github.com/giantswarm/aws-operator/pull/3119 and  https://github.com/giantswarm/cluster-operator/pull/1436 handling `v1alpha3`. I tested it in `gauss`.

### CODE REVIEW

File: pkg/aws/v1alpha3/common.go [hunk 1]
```diff
@@ -29,6 +29,9 @@ const (
 	// FirstHARelease is the first GS release for AWS that supports HA Masters
 	FirstHARelease = "11.4.0"
 
+	// FirstV1Alpha3Release is the first GS release for v1alpha3 GiantSwarm AWS CR's
+	FirstV1Alpha3Release = "16.0.0"
```
Comment (njuettner) [inline pkg/aws/v1alpha3/common.go:33]: First v1alpha3 release, we need to make a decision when we want to introduce it

**Review by anvddriesch — APPROVED** [review]
I did not check every single line but it looks good to me. The only thing is that we will likely not be able to roll back from v1alpha3 to v1alpha2, or at least the infraref would not get updated in the opposite direction, right?

**Comment by anvddriesch** [comment]
Ah, and a CHANGELOG entry would be good! :smile:

**Comment by njuettner** [comment]
> I did not check every single line but it looks good to me. The only thing is that we will likely not be able to roll back from v1alpha3 to v1alpha2, or at least the infraref would not get updated in the opposite direction, right?

CRD have been already rolled out for v1alpha3, those are the ones which are stored in etcd. We can get both right now, without specifying version you will get by default `v1alpha3` and if you want `v1alpha2` you need to specify the version, e.g. `k get awsclusters.v1alpha2.infrastructure.giantswarm.io`.

You're right `infraRef` will be updated to `v1alpha3` and as long you don't revert release version (e.g. `v16.0.0` back to `v15.x.x`) it wouldn't revert infraRef back to `v1alpha2`, but I don't think it's needed.

**Comment by anvddriesch** [comment]
Yeah I think it should be fine. I don't see why we would need to roll back.

**Pull request merged**